spec/requests/tasks_spec.rb

```diff
@@ -0,0 +1,51 @@
+describe 'TasksController' do
+  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+
+  def expect_deleted(*args)
+    args.each do |arg|
+      expect(MiqTask.find_by(:id => arg.id)).to be_nil
+    end
+  end
+
+  it 'deletes on DELETE' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete, :delete)
+
+    url = api_task_url(nil, task.id)
+
+    delete(url)
+
+    expect(response).to have_http_status(204)
+    expect_deleted(task)
+  end
+
+  it 'deletes on POST' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete)
+
+    url = api_task_url(nil, task.id)
```

Comment:
You may not need to assign this to a variable if you're only calling it once, same as above and below

Reply:
Sure I don't.

But is it not clearer to write..

```
url = function call with not-so clear params
data = create the structure

post(url, data)
```

than to have it all in one line, like

```ruby
  post(api_task_url(nil, task.id), {
      :action    => 'delete',
      :resources => [
        {:href => api_task_url(nil, task.id)},
        {:href => api_task_url(nil, task2.id)},
      ],
    })
```

But of course your repo, your rules, just tell me you're sure you want it that way :).

Reply:
My two penneth: I think "clearer" is always going to be a bit subjective in these cases. I personally find your counter example clearer.